gluten-core/src/main/scala/io/glutenproject/expression/DateTimeExpressionsTransformer.scala

```diff
@@ -56,6 +62,35 @@ class ExtractDateTransformer(
   }
 }
 
+
+class DateDiffTransformer(endDate: ExpressionTransformer, startDate: ExpressionTransformer,
+  original: DateDiff)
+  extends ExpressionTransformer with Logging {
+
+  override def doTransform(args: java.lang.Object): ExpressionNode = {
+    if (BackendsApiManager.getBackendName.equalsIgnoreCase(GlutenConfig.GLUTEN_VELOX_BACKEND)) {
+      throw new UnsupportedOperationException(s"not supported yet.")
+    }
+
+    // In Spark: datediff(endDate, startDate)
+    // In CH: date_diff('day', startDate, endDate)
+    val endDateNode = endDate.doTransform(args)
+    val startDateNode = startDate.doTransform(args)
+    val unitNode = ExpressionBuilder.makeStringLiteral("day")
```

Comment:
please check whether the velox backend need this parameter day

Reply:
It is needed too: https://facebookincubator.github.io/velox/functions/datetime.html#date_diff